### PR TITLE
test(client-debugger): Add test coverage for registry and audience

### PR DESCRIPTION
**Pull request by chentong7**
## Description
This code change is adding test coverage for client debugger registry events. It will verify debuggers added/removed from registry and audience.

## Log
~~~
  ClientDebugger unit tests
node_modules/.pnpm/mocha@10.2.0/node_modules/mocha/lib/reporters/spec.js:54
    ✔ Initializing debugger populates global (window) registry
node_modules/.pnpm/mocha@10.2.0/node_modules/mocha/lib/reporters/spec.js:76
    ✔ Validate multi-debugger contents are as expected
node_modules/.pnpm/mocha@10.2.0/node_modules/mocha/lib/reporters/spec.js:76
    ✔ Validate audience contents are as expected 
node_modules/.pnpm/mocha@10.2.0/node_modules/mocha/lib/reporters/spec.js:76
    ✔ Closing debugger removes it from global (window) registry and disposes it.
node_modules/.pnpm/mocha@10.2.0/node_modules/mocha/lib/reporters/spec.js:76
  4 passing (10ms)
~~~

### CODE REVIEW

File: packages/tools/client-debugger/client-debugger/src/test/ClientDebugger.test.ts [hunk 1]
```diff
@@ -26,9 +29,32 @@ describe("ClientDebugger unit tests", () => {
...
+	const registry: DebuggerRegistry = getDebuggerRegistry();
+
+	let clientId: string | undefined;
+
+	let debuggerRegistered = false;
+	let debuggerClosed = false;
+	let audienceAdded = false;
+	let audienceRemoved = false;
+
 	beforeEach(async () => {
```
Comment (Josmithr) [inline packages/tools/client-debugger/client-debugger/src/test/ClientDebugger.test.ts:41]: You'll want to add an `afterEach` hook to ensure all of the boolean state we're validating against is reset to `false` between tests.

**Review by Josmithr — CHANGES_REQUESTED** [review]
I left one comment about a minor issue with the test setup, otherwise this looks great! Thank you!

File: packages/tools/client-debugger/client-debugger/src/test/ClientDebugger.test.ts (hunk 1 above)
Comment (chentong7) [reply packages/tools/client-debugger/client-debugger/src/test/ClientDebugger.test.ts:41]: Added!

**Review by Josmithr — APPROVED** [review]
🚀

**Pull request merged**